File: java-code/src/classes/Manager.java
package classes;

import java.util.List;
import classes.base.Employee;
import runnables.ReadFile;
import runnables.WriteFile;

public class Manager extends FullTimeEmployee {

  public Manager(String firstName, String lastName, Position position, String dob, Department department,
                 Double salary){
    super(firstName, lastName, position, dob, department, salary);
  }

  // METHODS
  public void evaluateEmployee(Employee employee) {
    System.out.println("Evaluating: " + employee.GetFirstName());
    // TODO: Write actual code here
  }

  public void assignTask(Employee employee, String task) {
    System.out.println("Giving task: " + task + " to: " + employee.GetFirstName());
  }

  public void generateReport(List<Employee> list) {
    System.out.println("Generating Report");
    WriteFile writeFile = new WriteFile(list, "report.txt");
    writeFile.run();
  }

  public void getReport() {
    ReadFile readFile = new ReadFile("report.txt");
    readFile.run();
  }

  public void reportToManager() {
    System.out.println(GetFirstName() + " has no manager to report to!");
  }
}

File: c#-code/src/classes/Manager.cs
namespace APLCAssignment3.Classes
{
    using APLCAssignment3.Classes.Base;
    using APLCAssignment3.Runnables;
    using System.Collections.Generic;
    
    public class Manager : FullTimeEmployee
    {

        public Manager(string firstName, string lastName, Position position, string dob, 
                       Department department, double salary)
            : base(firstName, lastName, position, dob, department, salary)
        {}

        public void EvaluateEmployee(Employee employee)
        {
            Console.WriteLine($"Evaluating: {employee.GetFirstName()}");
        }

        public void AssignTask(Employee employee, string task)
        {
            Console.WriteLine($"Giving task: {task} to: {employee.GetFirstName()}");
        }

        public void GenerateReport(List<Employee> list)
        {
            Console.WriteLine("Generating Report");
            WriteFile writeFile = new WriteFile(list, "report.txt");
            writeFile.Run();
        }

        public void GetReport()
        {
            ReadFile readFile = new ReadFile("report.txt");
            readFile.Run();
        }

        public override void ReportToManager()
        {
            Console.WriteLine($"{GetFirstName()} has no manager to report to!");
        }
    }
}